### PR TITLE
Issue 1285 - add feature for `micronaut-test-rest-assured` module of Micronaut Test

### LINKED ISSUE #1285: Micronaut Test REST-Assured feature
@graemerocher  [added a REST-Assured module to Micronaut Test](https://github.com/micronaut-projects/micronaut-test/pull/591). 

We should add a feature which adds the `micronaut-test-rest-assured`

### PR DESCRIPTION
**Pull request by wetted**
closes #1285

### CODE REVIEW

**Comment by wetted** [comment]
The feature validation I implemented allows any language and only JUnit5 test framework. Does that make sense? Should we omit Groovy language, include KoTest? Let me know what you think and I'll make the changes if necessary.

File: starter-core/src/main/java/io/micronaut/starter/feature/test/MicronautRestAssured.java [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+    private static final String ARTIFACT_ID_MICRONAUT_TEST_REST_ASSURED = "micronaut-test-rest-assured";
+
+    @Override
+    public String getName() {
+        return "micronaut-test-rest-assured";
+    }
+
+    @Override
+    public String getTitle() {
+        return "Micronaut-Test Rest-Assured";
```
Comment (sdelamo) [inline starter-core/src/main/java/io/micronaut/starter/feature/test/MicronautRestAssured.java]: Replace `Rest-Assured` with `REST-assured`. That it is how they write it. see: https://rest-assured.io

File: starter-core/src/main/java/io/micronaut/starter/feature/test/MicronautRestAssuredValidator.java [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+import io.micronaut.starter.application.ApplicationType;
+import io.micronaut.starter.feature.Feature;
+import io.micronaut.starter.feature.validation.FeatureValidator;
+import io.micronaut.starter.options.Options;
+import jakarta.inject.Singleton;
+
+import java.util.Set;
+
+@Singleton
+public class MicronautRestAssuredValidator implements FeatureValidator {
```
Comment (sdelamo) [inline starter-core/src/main/java/io/micronaut/starter/feature/test/MicronautRestAssuredValidator.java]: remove `MicronautRestAssuredValidator`. 

You can use `micronaut-test-rest-assured` with Spock and KoTest. 

> You can then inject instances of [RequestSpecification](https://www.javadoc.io/static/io.rest-assured/rest-assured/5.1.0/io/restassured/specification/RequestSpecification.html) into test fields or method parameters 
(parameters are only supported with JUnit 5):

Based on the docs, I think you should be able to do: 

```groovy
@MicronautTest 
public class RestAssuredHelloWorldSpec extends Specification {
    @Inject  
   RequestSpecification requestSpec
   void "test hello world"() {
```

File: starter-core/src/test/groovy/io/micronaut/starter/feature/test/MicronautRestAssuredSpec.groovy [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+
+        then:
+        template.contains('testImplementation("io.micronaut.test:micronaut-test-rest-assured")')
+
+        where:
+        language << Language.values().toList()
+    }
+
+    @Unroll
+    void 'test gradle micronaut-test-rest-assured feature fails for language=#language when test framework is not Junit'() {
```
Comment (sdelamo) [inline starter-core/src/test/groovy/io/micronaut/starter/feature/test/MicronautRestAssuredSpec.groovy]: other test frameworks appart from Junit should be supported

**Review by sdelamo — CHANGES_REQUESTED** [review]

**Review by sdelamo — APPROVED** [review]

**Pull request merged**